app/src/main/java/ch/epfl/sdp/appart/favorites/FavoriteViewModel.java

```diff
@@ -66,4 +86,66 @@ public void initHome() {
     public MutableLiveData<List<Card>> getFavorites() {
         return lFavorites;
     }
+
+    /**
+     * Loads content from local database
+     */
+    private CompletableFuture<Void> localLoad() {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        localdb.getCards()
+                .exceptionally(e -> {
+                    result.completeExceptionally(e);
+                    return null;
+                })
+                .thenAccept(cards -> {
+                    lFavorites.setValue(cards);
+                    result.complete(null);
+                });
+        return result;
+    }

```

Comment:
Chain desired ? (just to make sure because we have a null pointer here too if it failed)